database/factories/ActieFactory.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace Database\Factories;
+
+use Illuminate\Database\Eloquent\Factories\Factory;
+use Illuminate\Support\Facades\DB;
+use App\Models\Actie;
+
+/**
+ * @extends \Illuminate\Database\Eloquent\Factories\Factory<\App\Models\Actie>
+ */
+class ActieFactory extends Factory
+{
+    
+    #Setting model
+    protected $model = Actie::class;
+    
+    # Setting dummy types
+    protected static $titles_map = [
+        'Klimaatdemo',
+        'Black Lives Matter',
+        'Kick Out Zwarte Piet',
+        'Woonprotest',
+        'Bio-industrie Protest'
+    ];
+
+    protected static $externe_link_map = [
+        'https://utrecht.nl',
+        'https://partijvoordedieren.nl',
+        'https://nijmegen.nl',
+        '#'
+    ] ;
+    
+    protected static $location_human_map = [
+
+        'De Dam, Amsterdam',
+        'Jaarbeursplein, Utrecht',
+        'Museumplein, Amsterdam',
+        'Valkhofpark, Nijmegen'
+    ];
+
+    protected static $slug_map = [
+        'bio-industrie-protest',
+        'black-lives-matter-amsterdam',
+        'black-lives-matter-nijmegen',
+        'black-lives-matter-utrecht',
+        'kick-out-zwarte-piet',
+        'klimaatdemo-amsterdam',
+        'klimaatdemo-nijmegen',
+        'klimaatdemo-utrecht',
+        'woonprotest-amsterdam'
+
+    ] ;
+    
+    protected static $status_map = [
+        'PUBLISHED',
+        'DRAFT',
+        'PENDING'        
+    ] ;
+
+
+    protected static $image_map = [
+        null,
+       'acties/pexels-karolina-grabowska-8106775-resize-500.jpg',
+       'acties/pexels-markus-spiske-3039036-resize-500.jpg',
+       'acties/pexels-paddy-o-sullivan-2369217-resize-500.jpg'
+   ];
+
+   protected static $keywords_map = [
+        null
+   ] ;
+
+   protected static $location_map = [
+    'POINT (5.1065183336479 52.088538312124)',
+    null    
+   ];
+
+    public function definition(): array
+    { 
+
+        #Picking up a semantic seed
+        $actie_name = $this->faker->randomElement(static::$titles_map);
+
+        #Picking up a location
+        $location_human = $this->faker->address();
+
+        #Extrating Street from location
+        preg_match("/^\D*(?=\d)/",$location_human, $m);
+        $pos = isset($m[0]) ? strlen($m[0]) : false;
+        $street = substr($location_human,0, $pos);
+        $street = str_replace(" ","",$street);
+        
+        #Calculating the first date
+        $start_date_obj = $this->faker->dateTimeBetween('-2year ',now()) ;
+        $start_date = $start_date_obj->format('Y-m-d');
+        $start_time = $start_date_obj->format('H:i:s');
+
+        # Calculating the end date from the start date
+        $end_date_obj =  $this->faker->dateTimeBetween( $start_date_obj,now()) ;
+        $end_date =  $end_date_obj ->format('Y-m-d');
+        $end_time =  $end_date_obj->format('H:i:s');
+
+        #Calculating create date from start date
+        $create_at_obj =  $this->faker->dateTimeBetween( "-2years",$start_date_obj) ;
```

Comment:
Why do you use '-2year' here? We want to generate actions in the future right?

Reply:
When I studied the first seed for this table, I noticed that the minimum year for start_date and create_at was 2022, 2 years ago.

So I did that:
  1- Took  the start_date from -2years until now(). So, we have start_date until the moment of seed. But I could set the range to picking dates to finishing in now() + 2years, so that we could have Acties with dates in the future. _Should I do it?_
 2- Picked up  the  create_at, being the date to creation of the record in the table,  always before the  start_date, limited to -2 years.